pkg/plugin/driver/job/job.go

```diff
@@ -18,45 +18,57 @@ package job
 
 import (
 	"bytes"
+	"fmt"
 	"time"
 
-	"github.com/heptio/sonobuoy/pkg/errlog"
-	"github.com/heptio/sonobuoy/pkg/plugin"
-	"github.com/heptio/sonobuoy/pkg/plugin/driver/utils"
 	"github.com/pkg/errors"
-	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	kuberuntime "k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/scheme"
+
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/heptio/sonobuoy/pkg/plugin"
+	"github.com/heptio/sonobuoy/pkg/plugin/driver/utils"
 )
 
 // Plugin is a plugin driver that dispatches a single pod to the given
 // kubernetes cluster
 type Plugin struct {
-	Definition      plugin.Definition
-	DfnTemplateData *plugin.DefinitionTemplateData
-	cleanedUp       bool
+	Definition plugin.Definition
+	SessionID  string
+	Namespace  string
+	cleanedUp  bool
 }
 
 // Ensure Plugin implements plugin.Interface
 var _ plugin.Interface = &Plugin{}
 
+type templateData struct {
```

Comment:
Most of this data is repeated, couldn't we push it up a layer?

Reply:
this was more of a boundary decision, I wanted every driver to have control over its own domain.